### PR TITLE
Regenerate rubygem-smart_proxy_discovery spec

### PR DESCRIPTION
**Pull request by evgeni**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.22
* [ ] 1.21
* [ ] 1.20

See Foreman's [plugin maintainer documentation](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

**Comment by evgeni** [comment]
I'll update this when https://github.com/theforeman/smart_proxy_discovery/commit/60adb369c01a37b575158b756a88b47ab3c7caa8 is in a release, as I want to get rid of the `rest-client < 2` dependency.

**Comment by evgeni** [comment]
this also includes the bump to 1.0.5

File: packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec [hunk 1]
```diff
@@ -71,19 +78,24 @@ mv %{buildroot}%{gem_instdir}/settings.d/discovery.yml.example \
...
 %dir %{gem_instdir}
+%config(noreplace) %attr(0640, root, foreman-proxy) %{foreman_proxy_settingsd_dir}/discovery.yml
+%license %{gem_instdir}/LICENSE
+%{gem_instdir}/bundler.d
 %{gem_libdir}
-%{gem_spec}
-%{foreman_proxy_bundlerd_dir}/discovery.rb
-%config(noreplace) %{foreman_proxy_settingsd_dir}/discovery.yml
-%doc %{gem_instdir}/LICENSE
+%{gem_instdir}/settings.d
```
Comment (ekohl) [inline packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec]: Probably not needed

File: packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec [hunk 2]
```diff
@@ -71,19 +78,24 @@ mv %{buildroot}%{gem_instdir}/settings.d/discovery.yml.example \
 
 %files
 %dir %{gem_instdir}
+%config(noreplace) %attr(0640, root, foreman-proxy) %{foreman_proxy_settingsd_dir}/discovery.yml
```
Comment (ekohl) [inline packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec]: Would it make sense to use `%{plugin_name}` here as well as in the %install of it?

File: packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec [hunk 3]
```diff
@@ -71,19 +78,24 @@ mv %{buildroot}%{gem_instdir}/settings.d/discovery.yml.example \
...
+%license %{gem_instdir}/LICENSE
+%{gem_instdir}/bundler.d
```
Comment (ekohl) [inline packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec]: Probably not needed

File: packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec (hunk 2 above)
Comment (evgeni) [reply packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec]: probably yes

**Comment by evgeni** [comment]
@ekohl all your comments seem correct -- but that's what the template generates right now. shall we fix the template first?

**Comment by ekohl** [comment]
That sounds like a good idea.

File: packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec (hunk 2 above)
Comment (evgeni) [reply packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec]: so this is generated in a loop https://github.com/theforeman/foreman-packaging/blob/360dc68181bea3c3684a2dff713c4cd880dc360d/gem2rpm/smart_proxy_plugin.spec.erb#L130-L133

which makes it not easy to replace with `%{plugin_name}` (as there might be more configs)

Comment (ekohl) [reply packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec]: Hmm ok, fair enough.

File: packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec (hunk 1 above)
Comment (evgeni) [reply packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec]: This comes directly from `gem2rpm`s `main_file_entries` which lists everything from the gems `files` list, so I'm not sure how to tell it not to add it here

File: packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec (hunk 3 above)
Comment (evgeni) [reply packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec]: same as `settings.d`

File: packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec (hunk 1 above)
Comment (evgeni) [reply packages/plugins/rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec]: I dropped these from this spec file for now.

**Pull request merged**